pkg/operation/botanist/component/istio/networkpolicies.go

```diff
@@ -0,0 +1,511 @@
+								// required for node local dns feature, allows egress traffic to CoreDNS
+								CIDR: fmt.Sprintf("%s/32", *values.DNSServerAddress),
+							},
+						})
+					}
+
+					if values.NodeLocalIPVSAddress != nil {
+						obj.Spec.Egress[0].To = append(obj.Spec.Egress[0].To, networkingv1.NetworkPolicyPeer{
+							IPBlock: &networkingv1.IPBlock{
+								// required for node local dns feature, allows egress traffic to node local dns cache
+								CIDR: fmt.Sprintf("%s/32", *values.NodeLocalIPVSAddress),
+							},
+						})
+					}
+
+					return nil
+				}
+			},
+		},
+		{
+			name: "allow-from-shoot-vpn",
+			transform: func(obj *networkingv1.NetworkPolicy) func() error {
+				return func() error {
+					obj.Annotations = map[string]string{
+						v1beta1constants.GardenerDescription: fmt.Sprintf("Allows Ingress from shoot vpn servers with label "+
+							"'%s=%s'", v1beta1constants.LabelApp, "vpn-shoot"),
+					}
+					obj.Spec = networkingv1.NetworkPolicySpec{
+						PodSelector: metav1.LabelSelector{
+							MatchLabels: map[string]string{
+								v1beta1constants.LabelApp: istiodAppLabelValue,
+							},
+						},
+						PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeIngress},
+						Ingress: []networkingv1.NetworkPolicyIngressRule{{
+							From: []networkingv1.NetworkPolicyPeer{{
+								PodSelector: &metav1.LabelSelector{
+									MatchLabels: map[string]string{
+										v1beta1constants.LabelApp:   "vpn-shoot",
+										v1beta1constants.GardenRole: v1beta1constants.GardenRoleSystemComponent,
+									},
+								},
+								NamespaceSelector: &metav1.LabelSelector{},
+							}},
+						},
+						},
+					}
+					return nil
+				}
+			},
+		},
+		{
+			name: "allow-from-aggregate-prometheus",
+			transform: func(obj *networkingv1.NetworkPolicy) func() error {
+				return func() error {
+					obj.Annotations = map[string]string{
+						v1beta1constants.GardenerDescription: fmt.Sprintf("Allows Ingress from pods with label "+
+							"'%s=%s'", v1beta1constants.LabelApp, "aggregate-prometheus"),
+					}
+					obj.Spec = networkingv1.NetworkPolicySpec{
+						PodSelector: metav1.LabelSelector{
+							MatchLabels: map[string]string{
+								v1beta1constants.LabelApp: istiodAppLabelValue,
+							},
+						},
+						PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeIngress},
+						Ingress: []networkingv1.NetworkPolicyIngressRule{{
+							From: []networkingv1.NetworkPolicyPeer{{
+								PodSelector: &metav1.LabelSelector{
+									MatchLabels: map[string]string{
+										v1beta1constants.LabelApp:   "aggregate-prometheus",
+										v1beta1constants.GardenRole: "monitoring",
+									},
+								},
+								NamespaceSelector: &metav1.LabelSelector{},
+							}},
+							Ports: []networkingv1.NetworkPolicyPort{{
+								Protocol: protocolPtr(corev1.ProtocolTCP),
+								Port:     func(port string) *intstr.IntOrString { v := intstr.FromString(port); return &v }("metrics"),
```

Comment:
Is a function here really necessary?

Would this not be sufficient?

```suggestion
								Port:     &intstr.FromString("metrics"),
```

Reply:
That doesn't work here.
It's not possible to get the address of a string literal. It needs to be assigned to a variable.

Reply:
Ok.